demo/lexer_demo.cpp

```diff
@@ -1,14 +1,111 @@
 #include <iostream>
+#include "fmt/core.h"
+#include "fmt/color.h"
+#include "strings.hpp"
 #include "lexer.hpp"
 #include "token.hpp"
+#include "strings.hpp"
 
-int main()
-{
-    std::cout << "Type\t\tPosition\t\tSrc\n";
-    std::vector<Token> tokens = lexer::read("./examples/ex4.rdd");
-    for (Token val : tokens)
-    {
-        std::cout << int(val.type) << "\t\t" << val.srcPos << "\t\t" << val.image << "\n";
+std::vector<std::string> g_TokTypeStr{
```

Comment:
These functions and that vector are ugly. I would like to avoid incorporating them into the codebase as much as possible.
And additionally, this file is _not_ a compiler. It's just a demo. So should not matter much.